src/types/roles.ts

```diff
@@ -0,0 +1,5 @@
+export type UserRolesObject = { [role: string]: boolean };
```

Comment:
You can do `Record<string, boolean>` but of course you don't get the label that way.